spock-core/core.gradle

```diff
@@ -80,3 +86,28 @@ task coreConsole(type: JavaExec,
     workingDir.mkdirs()
   }
 }
+
+def osgiPropertiesFile = file("$buildDir/verifyOSGiProperties.bndrun")
+
+// Bnd's Resolve task uses a properties file for its configuration. This
+// task writes out the properties necessary for it to verify the OSGi
+// metadata.
+tasks.register("osgiProperties", WriteProperties) {
+  outputFile = osgiPropertiesFile
+  property("-standalone", true)
+  property("-runee", "JavaSE-${javaVersion}")
+  property("-runrequires", "osgi.identity;filter:='(osgi.identity=${project.name})'")
+}
+
+// Bnd's Resolve task is what verifies that a jar can be used in OSGi and
+// that its metadata is valid. If the metadata is invalid this task will
+// fail.
+tasks.register("verifyOSGi", Resolve) {
```

Comment:
That verification task sounds really cool!

What should it detect? I'm asking, because I removed `/net.bytebuddy.*;resolution:=optional/,` from the config and it passes (the complete removal of the section what detected on the other hand).

Reply:
What the verification check does is:
a) ensure that the OSGi metadata of the bundle can actually be satisfied in OSGi with some set of dependencies (which also must run in OSGi)
b) determines if the dependencies required to run are available (ideally in the build)
c) if a) and b) succeeds it prints the _minimum_ set of bundles needed to run the bundle in OSGi

If you build with `--info` you will see the `verifyOSGi` task prints out the set of bundles required to run the non-optional aspects of spoke-core in OSGi.
e.g.
```
> Task :spock-core:verifyOSGi
...
-runbundles: assertj-core;version='[3.16.1,3.16.2)',groovy;version='[2.5.12,2.5.13)',junit-platform-commons;version='[1.7.0,1.7.1)',junit-platform-engine;version='[1.7.0,1.7.1)',junit-platform-launcher;version='[1.7.0,1.7.1)',junit-platform-testkit;version='[1.7.0,1.7.1)',org.opentest4j;version='[1.2.0,1.2.1)',spock-core;version='[2.0.0,2.0.1)'
```

Finally, when you removed `/net.bytebuddy.*;resolution:=optional/,` if you checked
a) **the metadata** - you would have noticed that bytebuddy was imported, just not optionally
b) **the result of `verifyOSGi`** - result of `-runbundles` probably included bytebuddy bundle :)
  in fact I check and this was the result:
```
-runbundles: assertj-core;version='[3.16.1,3.16.2)',groovy;version='[2.5.12,2.5.13)',junit-platform-commons;version='[1.7.0,1.7.1)',junit-platform-engine;version='[1.7.0,1.7.1)',junit-platform-launcher;version='[1.7.0,1.7.1)',junit-platform-testkit;version='[1.7.0,1.7.1)',net.bytebuddy.byte-buddy;version='[1.10.10,1.10.11)',org.opentest4j;version='[1.2.0,1.2.1)',spock-core;version='[2.0.0,2.0.1)'
```
:)

Reply:
I forgot another aspect of this verification is to help discover accidental, or incidental dependencies which caused transitive fan out to things you may not have expected just like your assertj scenario.